### PR TITLE
Make testDatafeedTimingStats_DatafeedJobIdUpdated test easier to debug

### PR DESCRIPTION
**Pull request by przemekwitek**
* Change test jobs/datafeeds names used by DatafeedTimingStats-related tests so that they do not clash with names from other tests
* Unroll the loops so that it is easier to track down which iteration failed should the exception occur

Related to: https://github.com/elastic/elasticsearch/issues/44196

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/packaging-sample

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java [hunk 1]
```diff
@@ -106,15 +108,15 @@ public void testDatafeedTimingStats_DatafeedRecreated() throws Exception {
...
 
-        String datafeedId = "lookback-datafeed";
+        String datafeedId = "lookback-datafeed-datafeed-recreated";
         DatafeedConfig datafeedConfig = createDatafeed(datafeedId, job.getId(), Arrays.asList("data"));
 
         registerJob(job);
         putJob(job);
 
-        for (int i = 0; i < 2; ++i) {
+        CheckedRunnable<Exception> process = () -> {
```
Comment (davidkyle) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java]: nit: `process` could be a more descriptive name -> `openAndRunJob`?

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

**Review by davidkyle — APPROVED** [review]
LGTM

"Closes #44196" is an optimistic statement, I'm not sure what the root cause it but the loop unrolling and unique names will help debugging and finding the failure in the log file. State is cleared between tests and jobs deleted so name clashes won't be problem (it is an error to create a job with the same name anyway).

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java]: Done.

**Review by przemekwitek — COMMENTED** [review]
I've removed the "Closes #issue" form the description.

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro

**Pull request merged**